Source/WebCore/page/LocalFrame.cpp

```diff
@@ -500,11 +500,14 @@ String LocalFrame::searchForLabelsBeforeElement(const Vector<String>& labels, El
     RefPtr<Node> n;
     for (n = NodeTraversal::previous(*element); n && lengthSearched < charsSearchedThreshold; n = NodeTraversal::previous(*n)) {
         // We hit another form element or the start of the form - bail out
-        if (is<HTMLFormElement>(*n) || (is<Element>(*n) && downcast<Element>(*n).isValidatedFormListedElement()))
+        if (is<HTMLFormElement>(*n))
+            break;
+
+        if (RefPtr element = dynamicDowncast<Element>(*n); element && element->isValidatedFormListedElement())
             break;
 
         if (n->hasTagName(tdTag) && !startingTableCell)
-            startingTableCell = downcast<HTMLTableCellElement>(n);
+            startingTableCell = uncheckedDowncast<HTMLTableCellElement>(n);
```

Comment:
I think it would be safer to just write `dynamicDowncast` here.

Reply:
Why not just keep downcast() since the expectation is that this will be the type?

Reply:
I guess there's something I don't understand. Given the tdTag check, why would this ever fail?

Reply:
It wouldn’t but the whole point is to have code that can be verified locally without understanding connections like the one between a tag and a DOM class. We aren’t trying to make “carefully written” code where we make no mistakes but rather obviously safe code.

Reply:
I see, and so only when it's a hot path would we consider using uncheckedDowncast? I should revert a change to uncheckedDowncast in another PR as well then.

Reply:
The check shouldn't fail but it's not necessarily obvious. I would also prefer to stay on the safe side and keep downcast<>() in this particular instance. Better safe than sorry.